girder_worker/plugins/docker/executor.py

```diff
@@ -157,20 +184,24 @@ def run(task, inputs, outputs, task_inputs, task_outputs, **kwargs):
             outputs['_stdout']['script_data'] = ''
             print_stdout = False
 
-    command = ['docker', 'run', '-u', str(os.getuid())]
+    uid = str(os.getuid())
+    gid = str(os.getgid())
+    command = ['docker', 'run', '-u', '%s:%s' % (uid, gid)]
 
     if tempdir:
-        command += ['-v', tempdir + ':/data']
+        command += ['-v', '%s:%s' % (tempdir, DATA_VOLUME)]
 
-    if 'entrypoint' in task:
-        command += ['--entrypoint', task['entrypoint']]
+    scripts_dir = os.path.join(
+        os.path.abspath(os.path.dirname(__file__)), 'scripts')
+    command += ['-v', '%s:%s:%s' % (scripts_dir, SCRIPTS_VOLUME, 'ro')]
```

Comment:
Sounds good. Makes sense. You can discard my comment.
On Apr 22, 2016 11:13 AM, "Zach Mullen" <notifications@github.com> wrote:

> In girder_worker/plugins/docker/executor.py
> <https://github.com/girder/girder_worker/pull/25#discussion_r60764415>:
>
> >
> > -    if 'entrypoint' in task:
> > -        command += ['--entrypoint', task['entrypoint']]
> > +    scripts_dir = os.path.join(
> > +        os.path.abspath(os.path.dirname(__file__)), 'scripts')
> > +    command += ['-v', '%s:%s:%s' % (scripts_dir, SCRIPTS_VOLUME, 'ro')]
>
> Using the scripts directory could allow us to easily expose other scripts
> that might be useful for containers in the future
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/girder/girder_worker/pull/25/files/a7ff4e5345a380c8a095fb1265ccaaaade746f93#r60764415>
>